source/_posts/2018-01-31-infrastrucutre-as-code-a-beginners-perspective.markdown

```diff
@@ -0,0 +1,123 @@
+
+**You might be thinking:** *Ok, that seems like good things to have, but can you please give me a concrete use case?*
+
+I will give you two 😁
+
+### Case 1 - Simple web app_1, app_2, app_3, ..., app_n
+
+Imagine you are working for a company that builds basic web apps for their clients, so, what does a basic web app need?
+
+Probably a web server and a database (I am oversimplifying), so every time you have a new client for a web app you go to AWS (or whatever cloud provider you use) and with some clicking around you get 2 VMs, one for running your app, and another one for the database.
+
+Oh wait, I forgot, you need two copies (at least) of this infrastructure, one for production and one for staging.
+
+Now you have two copies of the same infrastructure. You can start deploying your code for that client.
+
+Two months go by and you have another client that needs the same infrastructure, and there you go again, you click around in AWS and get that working.
+
+With IAC you can have a module that does all the heavy lifting for you, you run that code and in a couple of minutes you are good to go. You need to update all the infrastructure of your clients? No problem just update the module, run it and you are good to go!
+
+If you have been doing this by hand in AWS, then you are probably already seeing the benefits of having IAC. But if you don't believe me yet, just read the next case.
+
+### Case 2 - Going crazy with a complex app
+
+You came up with a great idea, a really cool web app, it started off simple, but now you have a lot of clients that rely on that app. In this case one web server and one database are not going to be enough.
+
+You start by adding one load balancer and another web server, you write the process down in a document. Some time goes by and you need to add another server, you follow the process in the document and you are done.
+
+Next month you realize that only one database server it not enough and you add a replica.
+
+Next you have to move some workload out of the main servers, you need a message queue and a server to run workers. You do it by hand, but at this point you forget to update the documents.
+
+Some months go by and you need to add more queues and workers, but since you forgot to add that into a document it is not easy to do it the exact same way. You do it, it works, but now the queues and workers have slightly different configurations, this is called *Configuration drift*.
+
+You decide to add some levels of caching in your app, you need Redis/Memcached, so you need to add more servers for those services. You also need to configure S3 to have your static files, a CDN to serve those worldwide, you need Route53 to configure your routes.
```

Comment:
After looking a quick look at Route53 documentation, I don't think that there is a concept called `route` (I mean, `route` seems to be used always as a verb, not as a noun).

What about `you need Route53 to configure DNS` instead of `you need Route53 to configure your routes`?

Anyway, I'm not 100% sure. You probably know much more since you were reading about the topic, so if you're sure that your original sentence is OK, go for it.